packages/agreement-process/test/agreementService.integration.test.ts

```diff
@@ -1434,4 +1473,131 @@ describe("Agreement service", () => {
       });
     });
   });
+  describe("archive agreement", () => {
+    it("should succeed when the requester is the consumer and the agreement is in a archivable state", async () => {
+      const authData = getRandomAuthData();
+      const eserviceId = generateId<EServiceId>();
+      const descriptorId = generateId<DescriptorId>();
+      const attributeId = generateId<AttributeId>();
+
+      const descriptor = buildDescriptorPublished(descriptorId, [
+        [buildEServiceAttribute(attributeId)],
+      ]);
+      const eservice = buildEService(eserviceId, authData.organizationId, [
+        descriptor,
+      ]);
+      const tenant = buildTenant(authData.organizationId);
+      const agreement = buildAgreement(
+        eserviceId,
+        tenant.id,
+        agreementState.active
+      );
+
+      await addOneEService(eservice, eservices);
+      await addOneTenant(tenant, tenants);
+      await addOneAgreement(agreement, postgresDB, agreements);
+
+      const archivedAgreementId = await agreementService.archiveAgreement(
+        agreement.id,
+        authData,
+        uuidv4()
+      );
+
+      await expectedAgreementArchived(
+        unsafeBrandId<AgreementId>(archivedAgreementId)
+      );
+    });
+
+    it("should throw a agreementNotFound error when the Agreement doesn't exist", async () => {
+      const authData = getRandomAuthData();
+      const eserviceId = generateId<EServiceId>();
+      const descriptorId = generateId<DescriptorId>();
+      const attributeId = generateId<AttributeId>();
+
+      const descriptor = buildDescriptorPublished(descriptorId, [
+        [buildEServiceAttribute(attributeId)],
+      ]);
+      const eservice = buildEService(eserviceId, authData.organizationId, [
+        descriptor,
+      ]);
+      const tenant = buildTenant(authData.organizationId);
+      const agreement = buildAgreement(
+        eserviceId,
+        tenant.id,
+        agreementState.active
+      );
+
+      await addOneEService(eservice, eservices);
+      await addOneTenant(tenant, tenants);
+      await addOneAgreement(agreement, postgresDB, agreements);
+
+      const agreementToArchiveId = generateId<AgreementId>();
```

Comment:
Same, no need to set-up all this data to trigger the agreementNotFound case, you could just call it with a random id without seeding any data (see #380 as example)

Reply:
Same also for the two other error cases below